draft-jaju-httpbis-zstd-window-size.md

```diff
@@ -50,33 +50,38 @@ informative:
 --- abstract
 
 Deployments of Zstandard, or "zstd", can use different window sizes to limit
-memory usage during compression and decompression. Some browsers and user agents
-limit window sizes to mitigate memory usage concerns, causing interoperability
-issues. This document updates the window size limit in RFC8878 from a
-recommendation to a requirement in HTTP contexts.
+memory usage during compression and decompression. Some browsers and user
+agents limit window sizes to mitigate memory usage concerns, causing
+interoperability issues. This document updates the window size limit in RFC8878
+from a recommendation to a requirement in HTTP contexts.
 
 
 --- middle
 
 # Introduction
 
 Zstandard, or "zstd", specified in {{?RFC8878}}, is a lossless data compression
-mechanism similar to gzip. When used with HTTP, the "zstd" Content Encoding
+mechanism similar to gzip. When used with HTTP, the "zstd" Content Coding
 token signals to the decoder that the content is Zstandard-compressed.
 
-Deployments of Zstandard can use different window sizes to configure the maximum
-memory a decoder requires to decompress a frame. Larger window sizes tend to
-improve the compression ratio, but cause more memory to be used. {{RFC8878}}
-provides a recommendation for decoders to support window sizes up to 8 MB, and
-for encoders to not generate frames requiring window sizes larger than 8 MB.
-However, it is just a recommendation ({{!RFC8878, Section 3.1.1.1.2}}).
+An important property of Zstandard-compressed content is its Window_Size
+({{!RFC8878, Section 3.1.1.1.2}}), which describes the maximum distance for
+matches and therefore how much of the content must be kept in memory during
+decompression. Larger Window_Size values tend to improve the compression ratio,
+but at the cost of increased memory usage.
 
 To protect against unreasonable memory usage, some browsers and user agents
-limit the maximum window size allowed. This causes incompatibilities if the
-content is compressed with a larger limit, leading to decreased interoperability.
+limit the maximum Window_Size they will handle. Since the "zstd" Content Coding
+does not specify a negotiation mechanism for the Window_Size to use, this
```

Comment:
```suggestion
limit the maximum Window_Size they will handle. This
```
This has caused some controversy in the WG in the past, so let's keep negotiation mechanisms out of scope here.